source/game/kart/KartObjectProxy.cc

```diff
@@ -9,15 +9,20 @@
 
 #include "game/system/RaceManager.hh"
 
+#include <algorithm>
+
 namespace Kart {
 
 KartObjectProxy::KartObjectProxy() : m_accessor(nullptr) {
-    if (s_list) {
-        s_list->append(this);
-    }
+    s_proxyList.push_back(this);
 }
 
-KartObjectProxy::~KartObjectProxy() = default;
+KartObjectProxy::~KartObjectProxy() {
+    auto iter = std::find(s_proxyList.begin(), s_proxyList.end(), this);
+    if (iter != s_proxyList.end()) {
+        s_proxyList.erase(iter);
+    }
```

Comment:
Because the list is temporary, we don't have to erase from it if we clear it every time at the start of `KartObject::Create`.

Reply:
Good point!